services/patient-greeting.js

```diff
@@ -0,0 +1,37 @@
+const express = require('express');
+
+const router = express.Router();
+
+// Middleware to verify all necessary data is available to process a response
+router.use((request, response, next) => {
+  const data = request.body;
+  if (!(data && data.prefetch && data.prefetch.patient && data.prefetch.patient.resource)) {
+    response.sendStatus(200);
```

Comment:
If there is no prefetch patient resource you return a 200?
Should this be a 400 (bad request)?
I think normally in this case the cds service would ideally fall back to retrieving the patient information itself, but since that isn't implemented and the expected request doesn't contain the information provided, this seems like a bad request.

Reply:
@mjhenkes Just checked the [latest spec](http://cds-hooks.org/#a-performance-tweak), and it looks like on failure to receive necessary data from prefetch, services should respond with a `HTTP 412 Precondition Failed status code`. I can make that change here. In a future enhancement we can add that logic to retrieve the patient information itself, if allowable. Thoughts?

Reply:
I want the CDS Service to retrieve Patient resource itself rather than returning a 412. The 412 status should be reserved for exceptional cases where the CDS Service is not capable of calling the FHIR server on its own.

Reply:
Do we need to go that far for this demo service? In a real cds services absolutely, but it might be overkill here.

Reply:
Given that the service we're replacing doesn't do this I'm fine if we log an issue and make that change later. I'm just sensitive to people looking at the code for this service and using the practices here in their real services.

The expectation of prefetching being optional has been a continual challenge for us because the Sandbox honors all prefetch requests.

Reply:
Cool, lets log an issue and make this change in a separate pull request.

Reply:
Issue logged: https://github.com/cds-hooks/sandbox-cds-services/issues/3